### DISCUSSION TITLE
Problemas com connection string e conexão

### DISCUSSION THREAD

**Discussion by pedro-mileipp** [balta-io/forum#194] [Tira Dúvidas] [answered]
- Boa tarde, prezado!
- Para fim de praticar um banco de dado para testar as funções de Create, Update e Delete ensinadas no módulo 2 de Dapper, criei um banco TestDapper com a seguinte tabela Products:
```sql
CREATE TABLE [Products](
[Id] UNIQUEIDENTIFIER NOT NULL,
[Nome] NVARCHAR(80) NOT NULL,
[Fabricante] NVARCHAR(80) NOT NULL,
[Quantidade] INT NOT NULL,
[VlUnitario] DECIMAL NOT NULL,
[Tipo] NVARCHAR(30) NOT NULL,
PRIMARY KEY(Id)
)
```
- Após fui fazer a conexão com o Dapper e foram dadas as seguintes mensagens de erro: 
<br>

` Unhandled exception. Microsoft.Data.SqlClient.SqlException (0x80131904): Cannot open database "TestDapper" requested by the login. The login failed.
Login failed for user 'sa'. `
` Error Number:4060,State:1,Class:11`
<br>

Essa é minha connection String: 
```csharp
const string connectionString = "Server=localhost,1433;Database=TestDapper;User ID=sa;Password=1q2w3e4r@#$; Trusted_Connection=False; TrustServerCertificate=True";
```

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @pedro-mileipp, como vai?

No seu código, você criou apenas uma tabela, acredito que faltou criar o banco `TestDapper` conforme aponta no erro.

```sql
CREATE DATABASE [TestDapper]
GO

USE [TestDapper]
GO

CREATE TABLE.....
```
